examples/decoding/plot_decoding_ssd_csp_eeg.py

```diff
@@ -0,0 +1,256 @@
+                                       cv=cv, n_jobs=1).std()
+###############################################################################
+# Let's visualize these new results
+fig, ax = plt.subplots(1, dpi=150, figsize=(8, 5))
+x_axis = np.arange(0, len(n_components), 1)
+ax.plot(x_axis, scores_ssd_csp_e, label='SSD+CSP')
+ax.fill_between(x_axis, scores_ssd_csp_e - std_ssd_csp_e,
+                scores_ssd_csp_e + std_ssd_csp_e, alpha=0.3)
+ax.yaxis.grid(True)
+ax.axhline(scores_csp, linestyle='-', color='k', label='CSP')
+ax.set_xlabel('number of components SSD')
+ax.set_ylabel('classification accuracy')
+ax.set_xticks(np.arange(0, len(n_components), 2))
+ax.set_xticklabels(np.arange(4, 65, steps * 2))
+ax.set_title('Impact of SSD after epoching')
+ax.legend()
+# As before, there is a huge impact in decoding performance with respect to
+# the number of components used in SSD.
+# Given that in this scenario SSD is applied after epoching, the edge effects
+# of filtering epoched data might explain why improvements values are lower.
+##############################################################################
+# Neurophysiological interpretation of the solution
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+# SSD can not only improve classification performance of the decoding pipeline,
+# as we showed, but it can also help towards a better interpretation of the
+# solution.
+# Here we are going to investigate the topographical plots of the CSP spatial
+# patterns with and without applying SSD before.
+
+# Just for the sake of this example, we are going to train CSP using all
+# available data. But, remember data should always be split into separate sets
+# to ensure the generalization capability of the model.
+##############################################################################
+# CSP
+# ^^^
+csp = CSP(n_components=4, log=True, norm_trace=False, reg='oas', rank='full')
+csp.fit(epochs_data, labels)
+# Plot topographies.
+pattern_epochs = EvokedArray(data=csp.patterns_[:4].T,
+                             info=raw_ssd.info)
+pattern_epochs.plot_topomap(units=dict(mag='A.U.'), time_format='')
+##############################################################################
+# SSD + CSP
+# ^^^^^^^^^
+
+ixd_max = np.argmax(scores_ssd_csp_e)
+n_comp_max = n_components[ixd_max]
+ssd = SSD(raw_ssd.info, filt_params_signal, filt_params_noise,
+          sort_by_spectral_ratio=True, return_filtered=True,
+          n_components=n_comp_max)
+
+ssd.fit(epochs_data)
+data_new = ssd.apply(epochs_data)
+csp.fit(data_new, labels)
+# Plot topographies.
+pattern_epochs = EvokedArray(data=csp.patterns_[:4].T,
+                             info=raw_ssd.info)
+pattern_epochs.plot_topomap(units=dict(mag='A.U.'), time_format='')
+
+# As it can be seen from the topographycal plots, the CSP patternes learned
+# after SSD was applied, enhances the left vs. right event related
+# (de)/synchronization patterns.
```

Comment:
```suggestion
# As it can be seen from the topographical plots, the CSP patterns that were
# learned after SSD was applied enhance the left vs. right event related
# (de-)synchronization patterns.
```